packages/ai-ide/src/browser/workspace-preferences.ts

```diff
@@ -38,6 +41,39 @@ export const WorkspacePreferencesSchema: PreferenceSchema = {
             items: {
                 type: 'string'
             }
+        },
+        [PROMPT_TEMPLATE_WORKSPACE_DIRECTORIES_PREF]: {
+            type: 'array',
+            title: nls.localize('theia/ai/promptTemplates/directories/title', 'Workspace-specific Prompt Template Directories'),
+            description: nls.localize('theia/ai/promptTemplates/directories/description',
+                'List of relative paths indicating folders in the current workspace to be scanned for WORKSPACE specific prompt templates. Templates from workspace directories \
+                have higher priority than global templates but lower priority than specific template files.'),
+            default: [],
+            items: {
+                type: 'string'
+            }
+        },
+        [PROMPT_TEMPLATE_WORKSPACE_EXTENSIONS_PREF]: {
+            type: 'array',
+            title: nls.localize('theia/ai/promptTemplates/extensions/title', 'Workspace-specific Prompt Template File Extensions'),
+            description: nls.localize('theia/ai/promptTemplates/extensions/description',
+                'List of file extensions in workspace-specific locations that are considered as prompt templates. When templates with the same ID exist in multiple locations, \
```

Comment:
It's mainly used for file extensions but our code actually does not care at all that these are file extensions. We just do an `endsWith` check. I like the code as it's more generic as just forcing extensions, however technically the preference description is too specific.

Reply:
I don't think so, we check for the extension:

```
 /**
     * Checks if the given file extension is registered as a prompt template extension
     * @param extension The file extension including the leading dot (e.g., '.prompttemplate')
     * @returns True if the extension is registered as a prompt template extension
     */
    protected isPromptTemplateExtension(extension: string): boolean {
        return this.templateExtensions.has(extension);
    }
```

And the callers:

`this.isPromptTemplateExtension(fileURI.path.ext)`

Reply:
I overlooked that, I just saw the `endsWith` in the file extension removal. :+1: